api/github_service/routes/Leaderboard.js

```diff
@@ -0,0 +1,35 @@
+const express = require('express');
+const router = express.Router();
+const fetch = require('node-fetch');
+const {
+  CLIENT_ID,
+  CLIENT_SECRET } = require('../../config/config.json').githubAppKeys;
+const {
+  OK,
+  NOT_FOUND
+} = require('../../util/constants').STATUS_CODES;
+
+/**
+ * Call to Github API to retrieve a repository's list of
+ * contributors and their commit count
+ */
+Router.get('/', async (req, res) => {
+  const { repository } = req.body;  // Repository name
+  fetch(`https://api.github.com/repos/SCE-Development/${req.body.repository}
```

Comment:
consider moving all of the github API logic into a file in `util/`, you can define a class like:

```js
class SceGithubApiHandler {
  // you could send in 2 params for id and secret
  constructor(clientId, clientSecret) {
    this.clientId = clientId;
    this.clientSecret = clientSecret;
  }

  // gets all the contributors for a given repo
  getContrubutorsForRepo(repoName) {
    fetch(`https://api.github.com/repos/SCE-Development/${repoName})
    // etc...
  }

  getPullRequestsFromRepo(repoName) {
    fetch(`https://api.github.com/repos/SCE-Development/${repoName}
  /pulls?client_id=${this.clientId}&client_secret=${this.clientSecret}`)
  }
}
```
a similar pattern can be found in [` api/github_service/util/SceGoogleApiHandler.js`](https://github.com/SCE-Development/Core-v4/blob/dev/api/google_api/util/SceGoogleApiHandler.js), the class is used in [`api/google_api/routes/Calendar.js`](https://github.com/SCE-Development/Core-v4/blob/8eef1002a09b7d37bda93abbf4a5dec55860dff4/api/google_api/routes/Calendar.js#L12)